### PR TITLE
[Move] Heal Block

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes?
This implements Heal Block and provides a test for it. 
Link to Old PR: https://github.com/pagefaultgames/pokerogue/pull/3047 

## Why am I doing these changes?
I think it would be nice for it to be implemented. 

## What did change?
- HealBlockTag (Duration: 5 Turns)
- PokemonHealPhase now accounts for HealBlockTag 
- Not implemented yet: moves restricted by heal block 
- Partially implemented: Tests 
- Lines for Localization 

Waiting for this implementation to finish up. 

### Screenshots/Videos
https://github.com/user-attachments/assets/08b71f5d-3325-437d-b4d4-975b7f803973

## How to test the changes?
Use Heal Block during a Pokemon battle and see if the affected pokemon can heal itself. 

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/fr/battle.ts [hunk 1]
```diff
@@ -155,5 +155,7 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} est blessé\npar la capacité {{moveName}} !",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} sacrifie des PV\net lance une malédiction sur {{pokemonName}} !",
   "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} est touché par la malédiction !",
-  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!"
+  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!",
+  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} can't restore its HP!",
+  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} can restore its HP again!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:160]:
```suggestion
  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} utilise\nla capacité Stockage {{stockpiledCount}} fois !",
  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} ne peut pas guérir !",
  "battlerTagsHealBlockOnRemove": "Le blocage de soins qui affectait\n{{pokemonNameWithAffix}} s’est dissipé !",
```

File: src/locales/de/battle.ts [hunk 2]
```diff
@@ -155,5 +155,7 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} wurde durch {{moveName}} verletzt!",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} nimmt einen Teil seiner KP und legt einen Fluch auf {{pokemonName}}!",
   "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} wurde durch den Fluch verletzt!",
-  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!"
+  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:158]:
```suggestion
  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} hortet {{stockpiledCount}}!",
```

File: src/locales/de/battle.ts [hunk 3]
```diff
@@ -155,5 +155,7 @@ export const battle: SimpleTranslationEntries = {
...
+  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} can't restore its HP!",
+  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} can restore its HP again!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:160]:
```suggestion
  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} kann nicht geheilt werden, da die Heilung blockiert wird!",
  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} kann wieder geheilt werden!",
```

File: src/locales/pt_BR/battle.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export const battle: SimpleTranslationEntries = {
   "trainerGo": "{{trainerName}} escolheu {{pokemonName}}!",
   "switchQuestion": "Quer trocar\nde {{pokemonName}}?",
   "trainerDefeated": "Você derrotou\n{{trainerName}}!",
-  "moneyWon": "Você ganhou\n₽{{moneyAmount}} por vencer!",
+  "moneyWon": "Você ganhou\n₽{{moneyAmount}} por ganhar!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:16]:
```suggestion
  "moneyWon": "Você ganhou\n₽{{moneyAmount}} por vencer!",
```

The correct word is "vencer" and it has been fixed earlier

File: src/locales/pt_BR/battle.ts [hunk 5]
```diff
@@ -156,4 +156,6 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} cortou seus PS pela metade e amaldiçoou {{pokemonName}}!",
   "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} foi ferido pelo Curse!",
   "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} estocou {{stockpiledCount}}!",
+  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} can't restore its HP!",
+  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} can restore its HP again!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:160]:
```suggestion
  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} não pode restaurar seus PS!",
  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} pode restaurar seus PS novamente!",
```

**Comment by EnochG1** [comment]
@frutescens
When `battlerTagsHealBlock` message is triggered?
In og game, there's three messages.

{{pokemonNameWithAffix}} was prevented\nfrom healing! // when heal block used
{{pokemonNameWithAffix}} was prevented\nfrom healing due to Heal Block! // when heal blocked pokemon try using heal
{{pokemonNameWithAffix}}'s Heal Block\nwore off! // when heal block turn over.

**Comment by mercurius-00** [comment]
@frutescens 
I noticed that some content in `It` and `zh-CN` was rolled back that shouldn't be within the scope of this PR. Please check if it's correct.

**Comment by 40chyan** [comment]
I was thinking that you are not using the newest beta branch as your pr base, causing many roll back commits in your pr, please refork a new branch with the newest version

**Review by 40chyan — CHANGES_REQUESTED** [review]
please use the newest beta branch as your pr base

File: src/locales/es/battle.ts [hunk 6]
```diff
@@ -155,5 +155,7 @@ export const battle: SimpleTranslationEntries = {
   "battlerTagsSaltCuredLapse": "{{pokemonNameWithAffix}} is hurt by {{moveName}}!",
   "battlerTagsCursedOnAdd": "{{pokemonNameWithAffix}} cut its own HP and put a curse on the {{pokemonName}}!",
   "battlerTagsCursedLapse": "{{pokemonNameWithAffix}} is afflicted by the Curse!",
-  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!"
+  "battlerTagsStockpilingOnAdd": "{{pokemonNameWithAffix}} stockpiled {{stockpiledCount}}!",
+  "battlerTagsHealBlock": "{{pokemonNameWithAffix}} can't restore its HP!",
+  "battlerTagsHealBlockOnRemove": "{{pokemonNameWithAffix}} can restore its HP again!",
```
Comment (AsdarDevelops) [inline src/locales/es/battle.ts:160]:
```suggestion
  "battlerTagsHealBlock": "¡{{pokemonNameWithAffix}} no puede restaurar sus PS!",
  "battlerTagsHealBlockOnRemove": "¡{{pokemonNameWithAffix}} ya puede recuperar PS!",
```

**Review by AsdarDevelops — COMMENTED** [review]
Added Spanish

**Pull request closed without merging**